packages/sanity/src/core/comments/context/comments/CommentsProvider.tsx

```diff
@@ -60,9 +65,50 @@ type TransactionId = string
 /**
  * @beta
  */
-export const CommentsProvider = memo(function CommentsProvider(props: CommentsProviderProps) {
-  const {children, documentId, documentType, isCommentsOpen, onCommentsOpen, sortOrder, type} =
-    props
+export function CommentsProvider(props: CommentsProviderProps) {
+  const parentContext = useContext(CommentsContext)
+  const documentId = props.documentId ?? parentContext?.documentId
+  const documentType = props.documentType ?? parentContext?.documentType
+  if (!documentId || !documentType) {
+    throw new Error(`Could not determine necessary comments provider values.`)
+  }
+
+  return (
+    <CommentsEnabledProvider documentId={documentId} documentType={documentType}>
+      <CommentsProviderInner {...props} />
+    </CommentsEnabledProvider>
+  )
+}
+
+const CommentsProviderInner = memo(function CommentsProviderInner(props: CommentsProviderProps) {
+  const parentContext = useContext(CommentsContext)
+  const {
+    children,
+    documentId = parentContext?.documentId,
+    documentType = parentContext?.documentType,
+    isCommentsOpen = parentContext?.isCommentsOpen,
+    onCommentsOpen = parentContext?.onCommentsOpen,
+    sortOrder = 'desc',
+    type = 'field',
+    getCommentLink = parentContext?.getCommentLink,
+    onClearSelectedComment = parentContext?.onClearSelectedComment,
+    selectedCommentId = parentContext?.selectedCommentId,
+    isConnecting = parentContext?.isConnecting ?? false,
+    onPathOpen = parentContext?.onPathOpen,
+  } = props
+
+  if (
+    !documentId ||
+    !documentType ||
+    isCommentsOpen === undefined ||
+    !onCommentsOpen ||
+    !getCommentLink ||
+    !onClearSelectedComment ||
+    !onPathOpen
```

Comment:
I don't have the full context on why we need to check if there is a parent comments context, but would it be possible to avoid throwing an error when optional props are missing? :)